### PR TITLE
Added proxy, because I needed it for our company.

### PR DESCRIPTION
**Pull request by ghost**
Use the Jenkins proxy settings if they are set to connect to internet.

### CODE REVIEW

**Comment by kellybyrd** [comment]
Do any of the other maintainers see any reason not to merge this? I can't test it right now and I'm pretty new to Jenkins plugin development so I don't get the reason for Hudson vs Jenkins classes (I know a bit of the history of the two projects, just not the reasons for using having the API split into a base and subclass), so I'm asking for opinions from the other maintainers.

I see only one nit with the code, the second-to-last closing brace added in the diff isn't indented correctly.

**Comment by chrisklaiber** [comment]
@kellybyrd I'm a bit new to Jenkins plugins myself, but from the Jenkins docs, it appears that `hudson.model.Hudson` exists for backwards compatibility (for use in Hudson maybe, but not sure) and inherits from `jenkins.model.Jenkins` which really defines the proxy configuration: http://javadoc.jenkins-ci.org/hudson/model/Hudson.html  Leaving it as `Hudson` is probably fine.

One naming nit: `getJenkinsClient()` should probably be `getHttpClient()`.

IMO, it's probably fine to leave as `Hudson` and merge.

File: src/main/java/jenkins/plugins/hipchat/StandardHipChatService.java [hunk 1]
```diff
@@ -48,6 +48,16 @@ public void publish(String message, String color) {
             }
         }
     }
+    
+    private HttpClient getJenkinsClient() {
+        HttpClient client = new HttpClient();
+        if (Hudson.getInstance() != null) {
```
Comment (kellybyrd) [inline src/main/java/jenkins/plugins/hipchat/StandardHipChatService.java]: Please change all uses of Hudson.getInstance() to Jenkins.getInstance().

File: src/main/java/jenkins/plugins/hipchat/StandardHipChatService.java [hunk 2]
```diff
@@ -48,6 +48,16 @@ public void publish(String message, String color) {
             }
         }
     }
+    
+    private HttpClient getJenkinsClient() {
```
Comment (kellybyrd) [inline src/main/java/jenkins/plugins/hipchat/StandardHipChatService.java]: Please change this to getHTTPClient()

**Pull request merged**